lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,78 @@
+import type { RenovateConfig } from '../../../config/types';
+import { get, getManagerList } from '../../../modules/manager';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: Set<string>;
+  managers: WorkerExtractConfig[];
+}
+
+function getFilteredManagerConfig(
+  config: RenovateConfig,
+  manager: string
+): WorkerExtractConfig {
+  // the type here is not completely correct because there isn't any defined type for managers
+  const managerConfig = config[manager] as WorkerExtractConfig;
+  const language = get(manager, 'language');
+  // the type here is not completely correct because there isn't any defined type for languages
+  const languageConfig = (
+    language && config[language] ? config[language] : {}
+  ) as WorkerExtractConfig;
+  const filteredConfig = {} as WorkerExtractConfig;
+
+  filteredConfig.manager = manager;
+  filteredConfig.npmrc = config.npmrc;
+  filteredConfig.npmrcMerge = config.npmrcMerge;
+
+  //  non-mergeable so manager level config is given preference
+  filteredConfig.enabled = managerConfig?.enabled ?? config.enabled;
+  filteredConfig.ignorePaths = managerConfig?.ignorePaths ?? [];
```

Comment:
Why aren't you using mergeChildConfig and then extracting out from the result?

Reply:
did try once but rejected the idea after had to use `any` type. But it is better to use it than repeating the implementation.